pytorch_lightning/trainer/training_loop.py

```diff
@@ -754,7 +754,11 @@ def training_forward(self, batch, batch_idx, opt_idx, hiddens):
             gpu_id = 0
             if isinstance(self.data_parallel_device_ids, list):
                 gpu_id = self.data_parallel_device_ids[0]
-            batch = self.transfer_batch_to_gpu(copy.copy(batch), gpu_id)
+
+            # Don't copy the batch since there is a single gpu that the batch could
+            # be referenced from and if there are multiple optimizers the batch will
+            # wind up copying it to the same device repeatedly.
+            batch = self.transfer_batch_to_gpu(batch, gpu_id)
```

Comment:
what happens with multiple GPUs?

Reply:
This is in the "self.single_gpu" condition, so I don't think it's possible to have multiple gpus here? But I don't know this code very well at all.